metricflow/dataflow/builder/node_evaluator.py

```diff
@@ -406,6 +407,10 @@ def evaluate_node(
         logger.debug(f"Candidate spec set is:\n{mf_pformat(candidate_spec_set)}")
 
         data_set_linkable_specs = candidate_spec_set.linkable_specs
+        # Look for which nodes can satisfy the linkable specs at their base grains. Custom grains will be joined later.
+        required_linkable_specs_with_base_grains = [
+            spec.with_base_grain if isinstance(spec, TimeDimensionSpec) else spec for spec in required_linkable_specs
```

Comment:
I just realized this with_base_grain thing isn't a function. Is this a property? If so, what is it doing? It looks like it's creating a new spec class with some property set, but if that's the case can't we have this be a property accessor defined directly? It doesn't seem like it saves us from any further instance checking/type refinement operations down the line.

Or is this switching the dimension grain from a custom one to the base grain? If so, do we need to do that, or should we just point to the `granularity.base_grain` or whatever the properties are called where needed?

Reply:
Update - we just looked at the definition in the branch and I think it'd be a good idea to make with_base_grain a function, it's more obvious that we're converting the spec into a different spec rather than accessing some granularity property or something.

Separately it'd be nice to not have to do this type refinement but I think it's fine here for now.

Reply:
Updated the property to be a function!
I would also love to not do the type refinement so maybe if we come up with a better solution today that can be removed.